Sources/GRPC/GRPCPayload.swift

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import NIO
+
+public protocol GRPCPayload {
+    init()
+}
+
+extension GRPCPayload {
+    init(serializedByteBuffer: inout NIO.ByteBuffer) throws {
+        self.init()
+    }
+    
+    func serialize(into buffer: inout NIO.ByteBuffer) throws {}
+}
```

Comment:
This is partly where the problem is coming from, this should be:

```swift
public protocol GRPCPayload {
  init(serializedByteBuffer: inout NIO.ByteBuffer) throws
  func serialize(into buffer: inout NIO.ByteBuffer) throws
}
```

There _should not_ be a default implementation on the `GRPCPayload`.